modules/gdscript/gdscript.cpp

```diff
@@ -1124,17 +1124,29 @@ void GDScriptInstance::get_property_list(List<PropertyInfo> *p_properties) const
 }
 
 void GDScriptInstance::get_method_list(List<MethodInfo> *p_list) const {
-
 	const GDScript *sptr = script.ptr();
 	while (sptr) {
 
 		for (Map<StringName, GDScriptFunction *>::Element *E = sptr->member_functions.front(); E; E = E->next()) {
-
 			MethodInfo mi;
+			GDScriptFunction *func =  E->get();
 			mi.name = E->key();
+
 			mi.flags |= METHOD_FLAG_FROM_SCRIPT;
-			for (int i = 0; i < E->get()->get_argument_count(); i++)
-				mi.arguments.push_back(PropertyInfo(Variant::NIL, "arg" + itos(i)));
+
+			// Get the arguments of the current method
+			for (int i = 0; i < func->get_argument_count(); i++) {
+				mi.arguments.push_back(PropertyInfo(func->get_argument_type(i).builtin_type, func->get_argument_name(i)));
+			}
+			// TODO: return the default arg values rather than their types
+			// Get the default args for the current method
+			for(int i = 0; i < func->get_default_argument_count(); i++) {
+				mi.default_arguments.push_back(func->get_default_argument(i));
```

Comment:
The issue I see here is that `get_default_argument()` doesn't really return the actual value. It has the same implementation as `get_default_argument_addr()` (i.e. it returns the address to the code that creates the value). So this can be a bit tricky to implement properly, since default values aren't available at this point.

Other than that this seems okay. But I don't think we should merge before the function is fixed, otherwise it might give even more confusing results.